src/main.rs

```diff
@@ -96,7 +96,19 @@ fn main() {
                 }
             }
         }
-
+        if debugger.reset_triggered {
+            debugger.reset_triggered = false;
+            let new_game: Option<Game> = opts.rom.as_ref().and_then(|romname| {
+                Game::new(romname.clone(), context.joypad.clone(), true).map_or_else(
+                    |e| {
+                        log::error!("while creating game context for {}: {:?}", romname, e);
+                        None
+                    },
+                    Option::Some,
+                )
+            });
+            game = new_game;
```

Comment:
Why not store directly the new_game in `game` ?

Reply:
I don't understand, that not what we're doing storing directly the `new_game` in `game` with `game = new_game` ?
or you want to move the code to a method of `GameContext` ?

Reply:
You are right. It comes from a time when I was using game to get the joypad before I realized that it wasn’t possible and that the joypad was stored in the context too.

Reply:
I think he meant game = Game::new

Reply:
> I think he meant game = Game::new
yes